lib/connect.js

```diff
@@ -234,29 +244,39 @@ const getTasks = (model, setupListeners) => {
         setupListeners(mongoClient);
       }
 
-      mongoClient.connect((err, _client) => {
-        ctx(err);
-
-        if (err) {
-          if (tunnel) {
-            debug('data-service connection error, shutting down ssh tunnel');
-            tunnel.close();
-          }
-
-          return cb(err);
+      /** @type {Promise<never>} */
+      const waitForTunnelError = new Promise(async(_, reject) => {
+        if (tunnel) {
+          const [error] = await once(tunnel, 'error');
+          reject(error);
         }
+      });
 
-        client = _client;
-
+      const closeTunnelOnError = async() => {
```

Comment:
isn't it better if this function takes `tunnel` as parameter rather than requiring a variable from the scope?

Reply:
Will change, sure

Reply:
Done in 851db7e